File: VB/BandedGridViewHideBands/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.Utils.Paint
Imports DevExpress.XtraGrid.Views.BandedGrid.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Drawing
Imports DevExpress.XtraGrid.Views.Grid.Drawing
Imports DevExpress.XtraGrid
Imports DevExpress.Utils.Drawing

Namespace BandedGridViewHideBands

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.DataSource = FillDataTable()
            Dim helper As MyPaintHelper = New MyPaintHelper(bandedGridView2, New GridBand() {gridBand3})
        End Sub

        Private Function FillDataTable() As DataTable
            Dim _dataTable As DataTable = New DataTable()
            _dataTable.Columns.Add(New DataColumn("bandedGridColumn1", GetType(String)))
            _dataTable.Rows.Add(New Object() {""})
            Return _dataTable
        End Function
    End Class
End Namespace

File: CS/BandedGridViewHideBands/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.Utils.Paint;
using DevExpress.XtraGrid.Views.BandedGrid.ViewInfo;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Drawing;
using DevExpress.XtraGrid.Views.Grid.Drawing;
using DevExpress.XtraGrid;
using DevExpress.Utils.Drawing;

namespace BandedGridViewHideBands
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = FillDataTable();
            MyPaintHelper helper = new MyPaintHelper(bandedGridView2, new GridBand[] { gridBand3 });
        }

        DataTable FillDataTable()
        {
            DataTable _dataTable = new DataTable();
            _dataTable.Columns.Add(new DataColumn("bandedGridColumn1", typeof(string)));
            _dataTable.Rows.Add(new object[] { "" });
            return _dataTable;
        }
    }
}
